### PR TITLE
IAP: Samples for Requesting ID Token

### PR DESCRIPTION
**Pull request by bshaffer**
Sample for fetching an ID Token as seen on [this documentation page](https://cloud.google.com/iap/docs/authentication-howto#authenticating_from_a_service_account).

Samples are dependent on https://github.com/googleapis/google-auth-library-ruby/pull/263. 

Tests will not pass on this CI until I get more info regarding testing.

Please let me know if I'm headed down the right track @dazuma !

### CODE REVIEW

File: iap/iap.rb [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  # project_id = "Your Google Cloud Project ID"
+  require "googleauth"
+
+  audience = nil
+  if project_number && project_id
+    # Expected audience for App Engine
+    audience = "/projects/#{project_number}/apps/#{project_id}"
+  elsif project_number && backend_service_id
+    # Expected audience for App Engine
+    audience = "/projects/#{project_number}/global/backendServices//#{backend_service_id}"
```
Comment (dazuma) [inline iap/iap.rb:60]: Does this really have a doubled slash?

Comment (bshaffer) [reply iap/iap.rb]: it sure doesn't... good catch. Fixed!

File: iap/iap.rb [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+def make_iap_request url:, client_id:
+  # [START iap_make_request]
+  # url = "The Identity-Aware Proxy-protected URL to fetch"
+  # client_id = "The client ID used by Identity-Aware Proxy"
+  require "googleauth"
+  require "faraday"
+
+  # The client ID as the target audience for IAP
+  id_token_creds = Google::Auth::Credentials.default \
+    target_audience: client_id
```
Comment (dazuma) [inline iap/iap.rb:26]: This can be put on one line.

File: iap/spec/iap_spec.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative "../iap"
+require "rspec"
+require "pp"
```
Comment (dazuma) [inline iap/spec/iap_spec.rb:17]: I don't think `pp` is used. However, you should `require "stringio"`.

File: iap/iap.rb [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+  headers = {}
+  id_token_creds.client.apply! headers
+
+  resp = Faraday.get url do |req|
+    req.headers = headers
+  end
```
Comment (dazuma) [inline iap/iap.rb:33]: This can be replaced by:
```ruby
resp = Faraday.get url, nil, headers
```

**Pull request merged**